projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.tsx

```diff
@@ -0,0 +1,207 @@
+    @Prop({ mutable: true }) value: string;
+    @Prop() name: string;
+    @Prop() fieldType: string;
+    @Prop() label: string;
+    @Prop() hint: string;
+    @Prop() keyPlaceholder: string;
+    @Prop() valuePlaceholder: string;
+    @Prop() required: boolean;
+    @Prop() requiredmessage: string;
+    @Prop() saveBtnLabel = 'Add';
+    @Prop() disabled = false;
+
+    @State() status: DotFieldStatus;
+    @State() keyValues: KeyValue[] = [];
+
+    @Event() valueChange: EventEmitter<DotFieldValueEvent>;
+    @Event() statusChange: EventEmitter<DotFieldStatusEvent>;
+
+    fieldInput: KeyValue = { key: '', value: '' };
+
+    /**
+     * Reset properties of the filed, clear value and emit events.
+     */
+    @Method()
+    reset(): void {
+        this.fieldInput = { key: '', value: '' };
+        this.keyValues = [];
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    @Listen('deleteKeyValue')
+    deleteKeyHandler(event: CustomEvent) {
+        this.keyValues = this.keyValues.filter((_item, internalIndex) => {
+            return internalIndex !== event.detail;
+        });
+        this.refreshStatus();
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    componentWillLoad(): void {
+        this.setInitialValue();
+        this.emitStatusChange();
+    }
+
+    hostData() {
+        return {
+            class: getClassNames(this.status, this.isValid(), this.required)
+        };
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    render() {
+        const labelTagParams: DotLabel = {
+            name: this.name,
+            label: this.label,
+            required: this.required
+        };
+        return (
+            <Fragment>
+                {getTagLabel(labelTagParams)}
+                <input
+                    class={getErrorClass(this.status.dotValid)}
+                    disabled={this.disabled || null}
+                    id={this.name}
+                    name="key"
+                    onInput={(event: Event) => this.setValue(event)}
+                    placeholder={this.keyPlaceholder}
+                    type="text"
+                    value={this.fieldInput.key}
+                />
+                <input
+                    class={getErrorClass(this.status.dotValid)}
+                    disabled={this.disabled || null}
+                    name="value"
+                    onInput={(event: Event) => this.setValue(event)}
+                    placeholder={this.valuePlaceholder}
+                    type="text"
+                    value={this.fieldInput.value}
+                />
+                <button
+                    type="button"
+                    disabled={this.disabled || null}
+                    onClick={() => this.addKey()}
+                >
+                    {this.saveBtnLabel}
+                </button>
+                {this.keyValues.length ? (
```

Comment:
If you move this to their own function, you don't need to disable-next-line:cyclomatic-complexity